libs/core/algorithms/include/hpx/parallel/algorithms/uninitialized_relocate.hpp

```diff
@@ -0,0 +1,275 @@
+
+            template <typename ExPolicy, typename Iter, typename FwdIter2>
+            static util::detail::algorithm_result_t<ExPolicy,
+                util::in_out_result<Iter, FwdIter2>>
+            parallel(
+                ExPolicy&& policy, Iter first, std::size_t count, FwdIter2 dest)
+            {
+                return parallel_uninitialized_relocate_n(
+                    HPX_FORWARD(ExPolicy, policy), first, count, dest);
+            }
+        };
+        /// \endcond
+
+        /////////////////////////////////////////////////////////////////////////////
+        // uninitialized_relocate
+        /// \cond NOINTERNAL
+        template <typename IterPair>
+        struct uninitialized_relocate
+          : public algorithm<uninitialized_relocate<IterPair>, IterPair>
+        {
+            constexpr uninitialized_relocate() noexcept
+              : algorithm<uninitialized_relocate, IterPair>(
+                    "uninitialized_relocate")
+            {
+            }
+
+            // non vectorized overload
+            template <typename ExPolicy, typename InIter, typename FwdIter2,
+                HPX_CONCEPT_REQUIRES_(
+                    hpx::is_sequenced_execution_policy_v<ExPolicy>)>
+            static util::in_out_result<InIter, FwdIter2> sequential(
+                ExPolicy&& policy, InIter first, InIter last, FwdIter2 dest)
+            {
+                auto count = std::distance(first, last);
+
+                return hpx::parallel::util::uninit_relocate_n(
+                    HPX_FORWARD(ExPolicy, policy), first, count, dest);
+            }
+
+            template <typename ExPolicy, typename InIter, typename FwdIter2>
+            static util::detail::algorithm_result_t<ExPolicy,
+                util::in_out_result<InIter, FwdIter2>>
+            parallel(
+                ExPolicy&& policy, InIter first, InIter last, FwdIter2 dest)
+            {
+                auto count = std::distance(first, last);
+
+                return parallel_uninitialized_relocate_n(
+                    HPX_FORWARD(ExPolicy, policy), first, count, dest);
+            }
+        };
+        /// \endcond
+
+    }    // namespace detail
+}    // namespace hpx::parallel
+
+namespace hpx {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::uninitialized_relocate_n
+    inline constexpr struct uninitialized_relocate_n_t final
+      : hpx::detail::tag_parallel_algorithm<uninitialized_relocate_n_t>
+    {
+        // clang-format off
+        template <typename InIter, typename Size,
+            typename FwdIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator_v<InIter> &&
+                hpx::traits::is_forward_iterator_v<FwdIter> &&
+                std::is_integral_v<Size>
+            )>
+        // clang-format on
+        friend FwdIter tag_fallback_invoke(hpx::uninitialized_relocate_n_t,
+            InIter first, Size count,
```

Comment:
Under `libs/core/algorithms/include/hpx/parallel/algorithms/uninitialized_relocate.hpp` the API should accept the same type of `first` and `last` iterator. The `InIter`, `Sent` API is expected to be in the ranges counterpart udner container_algorithms.